mountpoint-s3/scripts/fs_cache_bench.sh

```diff
@@ -0,0 +1,198 @@
+fi
+
+if [[ -n "${JOB_NAME_FILTER}" ]]; then
+  echo "Will only run fio jobs which match $JOB_NAME_FILTER"
+fi
+
+base_dir=$(dirname "$0")
+project_dir="${base_dir}/../.."
+cd ${project_dir}
+
+results_dir=results
+runtime_seconds=30
+startdelay_seconds=30
+iterations=10
+
+rm -rf ${results_dir}
+mkdir -p ${results_dir}
+
+run_fio_job() {
+  job_file=$1
+  bench_file=$2
+  mount_dir=$3
+  log_dir=$4
+
+  job_name=$(basename "${job_file}")
+  job_name="${job_name%.*}"
+
+
+  echo -n "Running job ${job_name} for ${iterations} iterations... "
+
+  for i in $(seq 1 $iterations);
+  do
+    echo -n "${i};"
+    # we know each fio job will be running for exactly 1 minute from the job definitions.
+    # there must be something wrong if it takes longer than that.
+    set +e
+    timeout 300s fio --thread \
+      --output=${results_dir}/${job_name}_iter${i}.json \
+      --output-format=json \
+      --directory=${mount_dir} \
+      --filename=${bench_file} \
+      --eta=never \
+      ${job_file}
+    job_status=$?
+    set -e
+    if [ $job_status -ne 0 ]; then
+      tail -1000 ${log_dir}/mountpoint-s3-*
+      echo "Job ${job_name} failed with exit code ${job_status}"
+      exit 1
+    fi
+  done
+  echo "done"
+
+  # combine the results and find an average value
+  jq -n 'reduce inputs.jobs[] as $job (null; .name = $job.jobname | .len += 1 | .value += (if ($job."job options".rw == "read")
+      then $job.read.bw / 1024
+      elif ($job."job options".rw == "randread") then $job.read.bw / 1024
+      elif ($job."job options".rw == "randwrite") then $job.write.bw / 1024
+      else $job.write.bw / 1024 end)) | {name: .name, value: (.value / .len), unit: "MiB/s"}' ${results_dir}/${job_name}_iter*.json | tee ${results_dir}/${job_name}_parsed.json
+}
+
+should_run_job() {
+    job_file=$1
+    if [[ -n "${JOB_NAME_FILTER}" ]]; then
+      if [[ "${job_file}" == *"${JOB_NAME_FILTER}"* ]]; then
+        # job name matches filter
+        return 0
+      else
+        # job name does not match filter
+        return 1
+      fi
+    fi
+}
+
+should_setup_storage() {
+    if [[ -n "${SKIP_STORAGE_SETUP}" ]]; then
+      echo "Skipping storage setup"
+      return 1
+    else
+      echo "Setting up storage"
+      return 0
+    fi
+}
+
+cache_benchmark () {
+  jobs_dir=mountpoint-s3/scripts/fio/read_cache
+
+  if should_setup_storage; then
+    local_storage=/localstorage
+    sudo mkfs -t ext4 /dev/nvme1n1 #/dev/nvme0n1 is used by root EBS vol
+    sudo mkdir $local_storage
+    current_user_id=$(id -u)
+    sudo mount /dev/nvme1n1 $local_storage
+    sudo chown -R $current_user_id $local_storage
+    echo "mounted local file system at $local_storage"
+  else
+    local_storage=/tmp
+  fi
+
+  for job_file in "${jobs_dir}"/*.fio; do
+
+    if ! should_run_job "${job_file}"; then
+      echo "Skipping job ${job_file} because it does not match ${JOB_NAME_FILTER}"
+      continue
+    fi
+
+    mount_dir=$(mktemp -d $local_storage/fio-XXXXXXXXXXXX)
```

Comment:
Here `mount_dir` is using the local storage right?

Reply:
Yeah. This is not a problem but I reverted it to use `/tmp` for the mount point.